spec/app/ics-interchain-queries/README.md

```diff
@@ -0,0 +1,151 @@
+---
+ics: TBA
+title: Interchain Query
+stage: draft
+category: IBC/APP
+requires: 25, 26, 27
+kind: instantiation
+author: Joe Schnetzler <schnetzlerjoe@gmail.com>
+created: 2022-01-06
+modified: 2022-02-12
+---
+
+## Synopsis
+
+This documents aims to document the structure, plan and implementation of the Interchain Queries Module allowing for cross-chain querying of state from IBC enabled chains.
+
+### Motivation
+
+Interchain Accounts (ICS-27) brings one of the most important features IBC offers, cross chain transactions (on-chain). Limited in this functionality is the querying of state from one chain, on another chain. Adding interchain querying via the Interchain Query module, gives unlimited flexibility to chains to build IBC enabled protocols around Interchain Accounts and beyond.
+
+### Definitions 
+
+- `Querying Chain`: The chain that is interested in getting data from another chain (Queried Chain). The Querying Chain is the chain that implements the Interchain Query Module into their chain.
+- `Queried Chain`: The chain that's state is being queried via the Querying Chain. The Queried Chain gets queried via a relayer utilizing its RPC client which is then submitted back to the Querying Chain.
+- `Key`: A Key in the querying chain module is a user specified query identifier for a query so a user can identify various query types. For example, if you were to query stakers/delegators on Gaia (Cosmos Hub) and store it in state. You can set the `key` as `stakers` and thus query via the key later on as needed.
+
+``Note:`` that a query in state can only have one unique key-id representation. For example, if a query with key = stakers & id = 0 is already in state, the query will fail if you try to add another query with key = stakers & id = 0. A query with key = stakers & id = 1 will succeed however (as long as that is not in state already).
+
+### Desired Properties
+
+- Permissionless: A Querying Chain can query a chain and implement cross-chain querying without any approval from a third party or chain governance.
+
+- Minimal Querying Chain Work: A Queried Chain has to do no implementation work or add any module to enable cross chain querying. By utilizing an RPC client on a relayer, this is possible.
+
+- Modular: Adding cross-chain querying should be as easy as implementing a module in your chain.
+
+- Control Queried Data: The Querying Chain should have ultimate control on how to handle queried data. Like querying for a certain query form/type.
+
+- Incentivization: In order to incentivize relayers for participating in interchain queries, a bounty is paid.
+
```

Comment:
In addition to desired properties, we should also in this case have a section to underscore what the querying module does not guarantee.

Should be clear to any users of querying module what **is** promised by interchain querying and what **is not** promised.

For example:

Since there's no prior negotiation, we cannot assume that data coming back from queried chain will be in an expected format.

We also cannot assume that the value stored under the key will be stable, since it may change after the query result has been submitted